MAUI/DataForm/Editors.md

```diff
@@ -0,0 +1,1147 @@
+
+#### Creating custom editor using IDataFormEditor 
+
+Views such as labels, buttons, and sliders can be loaded to custom editor. Here, entry is loaded as custom editor for `PhoneNumer` property. 
+
+#### Below mentioned example code for the Numeric editor which accepts only numbers 
+
+{% tabs %}
+{% highlight C# %}
+
+    dataForm.RegisterEditor("PhoneNumber", new NumericEditor(dataForm));
+    
+    public class NumericTextEditor : IDataFormEditor 
+    { 
+        private SfDataForm dataForm;
+
+            private DataFormCustomItem? dataFormCustomItem;
+
+    public NumericEditor(SfDataForm dataForm)
+    {
+        this.dataForm = dataForm;
+    }
+
+    public View CreateEditorView(DataFormItem dataFormItem) 
+    { 
+        Entry inputView = new Entry();
+        inputView.Keyboard = Keyboard.Numeric;
+        inputView.Placeholder = dataFormItem.PlaceholderText;
+        DataFormTextStyle textStyle = dataForm.EditorTextStyle;
+        inputView.TextColor = textStyle.TextColor;
+        inputView.FontSize = textStyle.FontSize;
+        inputView.FontFamily = textStyle.FontFamily;
+        inputView.FontAttributes = textStyle.FontAttributes;
+        inputView.TextChanged += this.OnViewTextChanged;
+        this.dataFormCustomItem = (DataFormCustomItem)dataFormItem;
+        this.dataFormCustomItem.EditorValue = string.Empty;
+        return inputView;
+
+            public void CommitValue(DataFormItem dataFormItem, View view)
+    {
+        if (view is InputView numericText)
+        {
+            double numericValue;
+            double.TryParse(numericText.Text, out numericValue);
+             dataFormItem.SetValue(numericValue);
+        }
+    }
+
+    private void ValidateValue(DataFormItem dataFormItem)
+    {
+        dataForm.Validate(new List<string>() { dataFormItem.FieldName });
+    }
+
+    private void OnViewTextChanged(object? sender, TextChangedEventArgs e)
+    {
+        if (sender is not InputView numericEntry || dataFormCustomItem == null)
+        {
+            return;
+        }
+
+        string? numericText = Regex.Replace(numericEntry.Text, "[^0-9]+", string.Empty);
+        if (numericText != numericEntry.Text)
+        {
+            numericEntry.Text = numericText;
+            return;
+        }
+
+        dataFormCustomItem.EditorValue = numericText;
+        this.ValidateValue(dataFormCustomItem);
+        this.CommitValue(dataFormCustomItem, numericEntry);
+    }
+
+    public void UpdateReadyOnly(DataFormItem dataFormItem)
+    {
+    }
+
+     
+{% endhighlight %}
+{% endtabs %}
+
+If `SfDataForm.ValidatetionMode` is set to `ValidationMode.Manual` then update `DataFormCustomItem.EditorValue` property once the value is committed to its respective model property. This value will be used for `DataFormValidateFormEventArgs.NewValues` on manual validation.
```

Comment:
If the `SfDataForm.ValidatetionMode` is set to `ValidationMode.Manual,` then update the `DataFormCustomItem.EditorValue` property once the value is committed to its respective model property. This value will be used for the `DataFormValidateFormEventArgs.NewValues` on manual validation.

Reply:
content updated